### ISSUE TITLE
Validation before rollback to stable release

### ISSUE THREAD

**Issue by sestrella** [stackbuilders/hapistrano#15]
Digging a little inside Capistrano source code, I didn't find any kind of validation mechanism that guarantees that you don't rollback to an incomplete revision. The rollback task seems that only changes the current release symlink to an existing revision and perform some rollback operations. Maybe we could use GitHub tags to perform the rollback, so once the deployment is done `hapistrano` should create a new tag with the timestamp we could use an approach like [this](http://casperfabricius.com/site/2009/09/20/manage-and-rollback-heroku-deployments-capistrano-style/).

Some other references:
- [Capistrano flow](http://capistranorb.com/documentation/getting-started/flow/)
- [Capistrano rake tasks](https://github.com/capistrano/capistrano/blob/master/lib/capistrano/tasks/deploy.rake)

**Comment by jsl** [comment]
Instead of modifying the github repo, could we do this:

* After all build steps, (but before symlinking?) touch release_name/BUILD_COMPLETE
* In rollback, only select directories that contain the file BUILD_COMPLETE.

Would that do the trick without modifying the git repo?

**Comment by mrkkrp** [comment]
@jsl I think this is a good idea. Does it make sense to add this to #52? Or should it be in a separate PR? Also, we could modify the algorthim that purges older releases to remove any incomplete releases before truncating the history.
